core/templates/pages/exploration-editor-page/services/router.service.spec.ts

```diff
@@ -193,79 +193,33 @@ describe('Router Service', () => {
     var jQuerySpy = spyOn(window, '$');
     jQuerySpy.withArgs('.oppia-editor-cards-container').and.returnValue(
       $(document.createElement('div')));
     jQuerySpy.and.callThrough();
+    spyOn($.fn, 'fadeOut').and.callFake(cb => {
+      cb();
+      $timeout.flush(200);
+      return null;
+    });
 
     expect(RouterService.getActiveTabName()).toBe('main');
     RouterService.navigateToMainTab('newState');
-    // To $watch the first $location.path call.
-    $rootScope.$apply();
-
-    // Function setTimeout is being used here to not conflict with
-    // $timeout.flush for fadeIn Jquery method. This first setTimeout is to wait
-    // the default time for fadeOut Jquery method to complete, which is 400
-    // miliseconds.
-    // Ref: https://api.jquery.com/fadeout/
-    setTimeout(() => {
-      // Waiting for $applyAsync be called, which can take ~10 miliseconds
-      // according to this ref: https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$applyAsync
-      setTimeout(() => {
-        expect(externalSaveSpy).toHaveBeenCalled();
-        expect(RouterService.getActiveTabName()).toBe('main');
-
-        $interval.flush(300);
-
-        expect(applyAsyncSpy).toHaveBeenCalled();
-        done();
-      }, 20);
-      $timeout.flush(150);
-    }, 400);
-  });
 
-  it('should not navigate to main tab when already there', done => {
-    var applyAsyncSpy = spyOn($rootScope, '$applyAsync').and.callThrough();
-
-    var jQuerySpy = spyOn(window, '$');
-    jQuerySpy.withArgs('.oppia-editor-cards-container').and.returnValue(
-      $(document.createElement('div')));
-    jQuerySpy.and.callThrough();
+    $rootScope.$apply();
+    flush();
 
+    expect(externalSaveSpy).toHaveBeenCalled();
     expect(RouterService.getActiveTabName()).toBe('main');
+
     RouterService.navigateToMainTab('newState');
-    // To $watch the first $location.path call.
+
     $rootScope.$apply();
+    flush();
 
-    // Function setTimeout is being used here to not conflict with
-    // $timeout.flush for fadeIn Jquery method. This first setTimeout is to wait
-    // the default time for fadeOut Jquery method to complete, which is 400
-    // miliseconds.
-    // Ref: https://api.jquery.com/fadeout/
-    setTimeout(() => {
-      // Waiting for $applyAsync be called, which can take ~10 miliseconds
-      // according to this ref: https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$applyAsync
-      setTimeout(() => {
-        expect(externalSaveSpy).toHaveBeenCalled();
-        expect(RouterService.getActiveTabName()).toBe('main');
-
-        $interval.flush(300);
-
-        expect(applyAsyncSpy).toHaveBeenCalled();
-
-        RouterService.navigateToMainTab('newState');
-        $timeout.flush();
-        $rootScope.$apply();
-
-        expect(RouterService.getActiveTabName()).toBe('main');
-        done();
-      }, 20);
-      $timeout.flush(150);
-    }, 400);
-  });
+    expect(RouterService.getActiveTabName()).toBe('main');
```

Comment:
Is there anything else we can also check like the `fadeout` call to ensure that it does not navigate to main tab when already there

Reply:
done